### PR TITLE
feat: BackgroundTaskDeferral for Android

### PR DESCRIPTION
**Pull request by pkar70**
GitHub Issue (If applicable): #

## PR Type

What kind of change does this PR introduce?
- Feature

## What is the current behavior?
No `BackgroundTaskDeferral` implementation, so in `App.OnBackgroundActivated` using `#if` is required.

## What is the new behavior?
 `BackgroundTaskDeferral` is implemented.
(although for Android it is empty implementation, as Android doesn't kill background tasks)

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [X] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [X] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information
See also #2655.

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UWP/ApplicationModel/Background/BackgroundTaskDeferral.Android.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+﻿
+#if __ANDROID__
```
Comment (MartinZikmund) [inline src/Uno.UWP/ApplicationModel/Background/BackgroundTaskDeferral.Android.cs:2]: You can remove the `#if` as long as the name of the file ends with `.Android.cs`

File: src/Uno.UWP/ApplicationModel/Background/BackgroundTaskDeferral.Android.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+
+namespace Windows.ApplicationModel.Background
+{
+
```
Comment (MartinZikmund) [inline src/Uno.UWP/ApplicationModel/Background/BackgroundTaskDeferral.Android.cs:11]: Remove the additional empty lines here and above

**Review by MartinZikmund — APPROVED** [review]

**Review by jeromelaban — APPROVED** [review]

File: src/Uno.UWP/ApplicationModel/Background/BackgroundTaskDeferral.Android.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+{
+	public partial class BackgroundTaskDeferral
+	{
+		/// <summary>
+		/// On Android, BackgroundTaskDeferral has no meaning (background tasks will not be cancelled by OS, unless it takes really much time)
+		/// </summary>
+		public void Complete()
+		{
+			// on Android, there is nothing to do
+		}
```
Comment (carldebilly) [inline src/Uno.UWP/ApplicationModel/Background/BackgroundTaskDeferral.Android.cs:15]: It's not the only platform where it has no meaning.

Comment (pkar70) [reply src/Uno.UWP/ApplicationModel/Background/BackgroundTaskDeferral.Android.cs:15]: Maybe. But I have no iOS device, no macOS device... I have (besides UWP) only Android device (now in servicing, as accumulator is dead). And while porting my UWP apps to Android, I'm forced to use

#if` NETFX_CORE
BackgroundTaskDeferral.Complete();
#endif

So this small addition makes porting of apps easier for me.

**Pull request merged**